### ISSUE TITLE
How to get the actual return code from bd_lvm_vgreduce() function

### ISSUE THREAD

**Issue by spictera** [storaged-project/libblockdev#689]
Hi,

I would like to retry if the device is busy. But are unable to understand how to get the actuall error code from the command.

```
                ret = bd_lvm_vgreduce(vg_name,snapshot->device,NULL,&error);
                if (!ret) {
                        // ignore code 5:   Physical volume "/dev/nbd3" still in use
                        if(error->code != 5) {
                                printf(msg,"%s (%s, %d)\n", error->message, g_quark_to_string (error->domain), error->code);
                        }
                }

```
But this doesn't work.

Suggestions?

Regards Tomas

**Comment by tbzatek** [comment]
Moved to `libblockdev` as indicated by the `bd_` symbol prefix.

The _exit code 5_ is a return code from the particular `lvm` comman call. However this is not always translated to a specific error in the `BD_LVM_ERROR` domain, like in this case.

**Comment by spictera** [comment]
Hi ThomasThanks for the replySo what is the best suggestion?to catch the error message, match the text using strcmp()? and then retry if there is a match?I tried to perform simpler way, by just retrying if there is an error, but that gives Glibc error as the error->xxx tries to expand the text.Why are you not using standard C routines where rc code is picked up by the caller, instead of a boolean?In that way it should be much simpler, as the caller can use the rc code to decide what action to perform anyhow, how can the bd_ function be called again?Do I have to release the error memory allocations before? and if so how?regards Tomas---- On Sat, 08 Jan 2022 12:06:09 +0100  ***@***.******@***.***> wrote ---- 
Moved to libblockdev as indicated by the bd_ symbol prefix. 
The exit code 5 is a return code from the particular lvm comman call. However this is not always translated to a specific error in the BD_LVM_ERROR domain, like in this case.
